assets/js/hocs/with-store-cart-api-hydration.js

```diff
@@ -13,13 +18,45 @@ import { useSelect } from '@wordpress/data';
  */
 const useStoreCartApiHydration = () => {
 	const cartData = useRef( getSetting( 'cartData' ) );
+	const { setIsCartDataStale } = useDispatch( CART_STORE_KEY );
 
 	useSelect( ( select, registry ) => {
 		if ( ! cartData.current ) {
 			return;
 		}
+		const { isResolving, hasFinishedResolution, isCartDataStale } = select(
+			CART_STORE_KEY
+		);
 
-		const { isResolving, hasFinishedResolution } = select( CART_STORE_KEY );
+		/**
+		 * This should only execute once. When the code further down the file executes
+		 * then the condition directly below this comment should never evaluate to true
+		 * on subsequent executions. Because of this localStorage.getItem won't be
+		 * called multiple times.
+		 */
+		if (
+			! isCartDataStale() &&
+			! isResolving( 'getCartData' ) &&
+			! hasFinishedResolution( 'getCartData', [] )
+		) {
+			const lastCartUpdateRaw =
+				window.localStorage.getItem( LAST_CART_UPDATE_TIMESTAMP_KEY ) ||
+				'';
+
+			const lastCartUpdate = parseFloat( lastCartUpdateRaw );
+			const cartGeneratedTimestamp = parseFloat(
+				cartData.current.generated_timestamp
+			);
+
+			const needsUpdateFromAPI =
+				! isNaN( cartGeneratedTimestamp ) &&
+				! isNaN( lastCartUpdate ) &&
+				lastCartUpdate > cartGeneratedTimestamp;
+
+			if ( needsUpdateFromAPI ) {
+				setIsCartDataStale();
+			}
+		}
```

Comment:
What works better here is to don't do `|| ''` for lastCartUpdateRaw. Test if there is a value and then do all the logic. This way we don't do unnecessary operations. 
```js
const lastCartUpdateRaw = window.localStorage.getItem( LAST_CART_UPDATE_TIMESTAMP_KEY );
        
if (lastCartUpdateRaw) {
        const lastCartUpdate = parseFloat( lastCartUpdateRaw );
		const cartGeneratedTimestamp = parseFloat(
		    cartData.current.generated_timestamp
	    );
        
	const needsUpdateFromAPI =
		! isNaN( cartGeneratedTimestamp ) &&
		! isNaN( lastCartUpdate ) &&
		lastCartUpdate > cartGeneratedTimestamp;
        
			        if ( needsUpdateFromAPI ) {
				        setIsCartDataStale();
			        }
}
			
```